### ISSUE TITLE
Double Matcheroo output making Triggers fire twice

### ISSUE THREAD

**Issue by Arcwin** [BeipDev/BeipMU#140]
While its likely something simple I've overlooked I'm having a problem of my triggers getting fired twice when the matched text is repeated on the screen.  

As an example, a simple trigger to make me keep foraging in a game:
Description: Forage1
Matcharoo: You found
Text to be sent: forage

Example of what the game outputs normally:  You found two parsnips.

What I see on my screen with the trigger active:
You found two parsnips.
You found two parsnips.

This makes the trigger fire twice naturally.  What I don't understand is why the output (with trigger active) is being displayed twice.

**Comment by BeipDev** [comment]
Hmm, let's see if we can narrow it down then. What are you sending to cause the trigger to first hit? Can you try changing the text sent to not be forage? Can you also look at the network debugger to see exactly what's being sent/received? (Help->Debuggers->Network debugger...)

**Comment by BeipDev** [comment]
(oh, and the Discord is great for helping to figure out trigger issues if that's an option for you!)

**Comment by Arcwin** [comment]
It does this with all of my triggers, not just 'forage'  I found the network debugger, and here is its output for a command 'jump'

 Sent 6 bytes
6A 75 6D 70 0D 0A 
jump CR LF 

 Received 58 bytes
59 6F 75 20 6A 75 6D 70 20 75 70 20 61 6E 64 20 64 6F 77 6E 2E 0D 0A 1B 5B 30 6D 1B 5B 33 32 6D 1B 5B 30 6D 20 1B 5B 31 6D 1B 5B 33 36 6D 3E 1B 5B 30 6D 20 1B 5B 30 6D FF F9 
You jump up and down. CR LF ESC [0m ESC [32m ESC [0m ESC [1m ESC [36m> ESC [0m ESC [0m TELNET_IAC TELNET_GA 

On my screen I see:
You jump up and down.



Then I made a trigger.  Only the following is set:

description: jumping
matcharoo: You jump up and down.
send text: say haaah!

 Sent 6 bytes
6A 75 6D 70 0D 0A 
jump CR LF 

 Received 58 bytes
59 6F 75 20 6A 75 6D 70 20 75 70 20 61 6E 64 20 64 6F 77 6E 2E 0D 0A 1B 5B 30 6D 1B 5B 33 32 6D 1B 5B 30 6D 20 1B 5B 31 6D 1B 5B 33 36 6D 3E 1B 5B 30 6D 20 1B 5B 30 6D FF F9 
You jump up and down. CR LF ESC [0m ESC [32m ESC [0m ESC [1m ESC [36m> ESC [0m ESC [0m TELNET_IAC TELNET_GA 

 Sent 24 bytes
73 61 79 20 68 61 61 61 68 21 0D 0A 73 61 79 20 68 61 61 61 68 21 0D 0A 
say haaah! CR LF say haaah! CR LF 

 Received 71 bytes
1B 5B 31 6D 1B 5B 33 37 6D 59 6F 75 20 65 78 63 6C 61 69 6D 2C 20 22 48 61 61 61 68 21 22 1B 5B 30 6D 0D 0A 1B 5B 30 6D 1B 5B 33 32 6D 1B 5B 30 6D 20 1B 5B 31 6D 1B 5B 33 36 6D 3E 1B 5B 30 6D 20 1B 5B 30 6D FF F9 
ESC [1m ESC [37mYou exclaim, "Haaah!" ESC [0m CR LF ESC [0m ESC [32m ESC [0m ESC [1m ESC [36m> ESC [0m ESC [0m TELNET_IAC TELNET_GA 

 Received 71 bytes
1B 5B 31 6D 1B 5B 33 37 6D 59 6F 75 20 65 78 63 6C 61 69 6D 2C 20 22 48 61 61 61 68 21 22 1B 5B 30 6D 0D 0A 1B 5B 30 6D 1B 5B 33 32 6D 1B 5B 30 6D 20 1B 5B 31 6D 1B 5B 33 36 6D 3E 1B 5B 30 6D 20 1B 5B 30 6D FF F9 
ESC [1m ESC [37mYou exclaim, "Haaah!" ESC [0m CR LF ESC [0m ESC [32m ESC [0m ESC [1m ESC [36m> ESC [0m ESC [0m TELNET_IAC TELNET_GA 


However on my screen I see:
You jump up and down.
You jump up and down.
 > You exclaim, "Haaah!"
 > You exclaim, "Haaah!"

maybe could get on discord sometime with an invitation, but need to arrange a time.

**Comment by BeipDev** [comment]
Hmm, it's definitely sending the trigger text twice. I did a simple test of what you had and it sends it only once for me. Is it possible you have a duplicate of the trigger in the global, server, or character sections?

The Discord invite is on the web page and in the Help->Welcome... screen. I'm around at most times, and other trigger experts are around the other times, but feel free to tell me when if you're planning on joining.

**Comment by BeipDev** [comment]
Looks like we figured out your issue.. for anyone else looking at this, the 'Invisible MUD Prompt Workaround' option was set on the server.

This option caused the received text to be processed by the trigger twice, once when the incomplete prompt was received and processed and a second time when the final version was processed.
